File: python_codes/longest_common_prefix.py
"""
Write a function to find the longest common prefix string amongst an array of strings.
If there is no common prefix, return an empty string "".
"""


def lcp(list1):
	pref = list1[0]
	pref_length = len(pref)
	
	for i in list1[1:]:
		while pref != i[0:pref_length]:
			pref_length -= 1

			if pref_length == 0:
				return ""
			
			pref = pref[0:pref_length]
	return pref


print(lcp(["flower","flow","flight"]))

File: javascript_codes/longestCommonPrefix.js
var longestCommonPrefix = function(strs) {
    let pref = strs[0]
    let prefLength = pref.length

    for(let i = 1;i<strs.length;i++){
    	let s = strs[i]
    	while(pref !== s.substring(0,prefLength)){
    		prefLength--

    		if(prefLength===0){
    			return ""
    		}

    		pref = pref.substring(0,prefLength)
    	}
    }
    return pref
};